src/iuse_actor.cpp

```diff
@@ -1131,28 +1132,105 @@ long firestarter_actor::use( player &p, item &it, bool t, const tripoint &spos )
     if( t ) {
         return 0;
     }
-
     tripoint pos = spos;
-    float light = light_mod( pos );
     if( !prep_firestarter_use( p, pos ) ) {
         return 0;
     }
-
+    int firestart_level = it.get_quality( quality_id( "FIRESTARTING" ) );
+    const time_point update_to = calendar::turn;
+    const time_point update_from = calendar::turn - DAYS( 1 );
+    // get the general wetness of this area from history of rain
+    bool sheltered = g->is_sheltered( pos );
+    auto accum_weather = sum_conditions( update_from, update_to, pos );
+    int wetness = 1;
+    if( !sheltered ) {
+        wetness =  std::max( 1, static_cast<int>( accum_weather.rain_amount / 7000 ) );
+    }
+    const oter_id &cur_om_ter = overmap_buffer.ter( g->m.getabs( pos ) );
+    const w_point weatherPoint = *g->weather_precise;
+    double windpower = weatherPoint.windpower;
+    windpower = get_local_windpower( windpower, cur_om_ter, pos,
```

Comment:
You declare `windopower` one line above, assign some value to it, and then assign some other value on the very next line.